### DISCUSSION TITLE
What is the best building process for creating an area of rooms (with all connections)?

### DISCUSSION THREAD

**Discussion by RodRitter** [evennia/evennia#2655] [Evennia Questions]
Good day all,

Just want to start out by saying I've fallen inlove with Evennia and enjoying it thoroughly.

I'm at the point where I want to build a grid-based "area" - let's say 5x5 - with all cardinal directions linked. First I tried using `@tunnel`, but found a problem when looping back around and had to resort to manually using `@open `and `@link `- which got very time consuming and not practical for larger environments.

My thought was to create a script that could run the process, given a few parameters, and automatically link them up. I couldn't figure out how toget the script to get a caller so I could use caller.msg, so here I am.

I'm wondering if:

- There's a catch to using `@tunnel` where I can easily hook up the rooms without jumping around, looking for room ID's (my rooms are "generic floor tiles" for my game, not uniquely named rooms)
- Could I run a script to generate rooms that I want? How do I go about doing that within the scope of the framework?

I'd appreciate any help.
Thanks in advance

**Comment by InspectorCaracal** [comment]
You might find [this contrib module](https://github.com/evennia/evennia/blob/master/evennia/contrib/mapbuilder.py) useful. I think it can do what you need right out of the box, but if not it should still be a good example to start from.

**Reply by davidrideout** [reply]
I switched to the dev branch for this very reason. The [`xyzgrid` contrib](https://www.evennia.com/docs/1.0-dev/Contribs/Contrib-XYZGrid.html) module does a lot of the grunt work (but it's extremely quirky).
